contracts/Market.sol

```diff
@@ -1349,16 +1352,26 @@ contract Market is
   }
 
   /**
-   * @dev Authorizes withdrawal for the removal. Reverts if the caller is not the owner of the removal and
-   * does not have the role `MARKET_ADMIN_ROLE`.
-   * @param owner The owner of the removal.
-   * @return Returns true if the caller is the owner, an approved spender, or has the role `MARKET_ADMIN_ROLE`,
-   * false otherwise.
+   * @dev Authorizes withdrawal for the removal. Reverts if the caller is not the supplier of the removal and
+   * does not have the role `MARKET_ADMIN_ROLE`, or if the recipient of the removal is either not the supplier
+   * of the removal or does not have the `Removal.CONSIGNOR_ROLE`.
+   * @param supplier The supplier of the removal.
+   * @return Returns true if the caller is the supplier, an approved spender, or has the role `MARKET_ADMIN_ROLE`,
+   * and the recipient is the removal supplier or has the `Removal.CONSIGNOR_ROLE`, false otherwise.
    */
-  function _isAuthorizedWithdrawal(address owner) internal view returns (bool) {
-    return (_msgSender() == owner ||
-      hasRole({role: MARKET_ADMIN_ROLE, account: _msgSender()}) ||
-      _removal.isApprovedForAll({account: owner, operator: _msgSender()}));
+  function _isAuthorizedWithdrawal(
+    address supplier,
+    address to
+  ) internal view returns (bool) {
+    return
+      (_msgSender() == supplier ||
+        hasRole({role: MARKET_ADMIN_ROLE, account: _msgSender()}) ||
+        _removal.isApprovedForAll({
+          account: supplier,
+          operator: _msgSender()
+        })) &&
+      (_removal.hasRole({role: _removal.CONSIGNOR_ROLE(), account: to}) ||
+        supplier == to);
```

Comment:
Might read a little cleaner to split these up into additional internal functions IE - `_isApprovedSpender || _hasMarketAdminRole`, etc...
Nit/nice-to-have, just hard to parse all these conditional statements.

Reply:
took the middle route by defining each predicate as its own well-named variable, but leaving the helper as one function

Reply:
That helps a lot, thanks!